Geometry/HcalTowerAlgo/test/HcalGeometryAnalyzer.cc

```diff
@@ -19,62 +22,65 @@ class HcalGeometryAnalyzer : public edm::one::EDAnalyzer<>
   void endJob() override {}
 
 private:
-  std::string m_label;
+  edm::ParameterSet ps0;
+  bool              useOld_;
 };
 
-HcalGeometryAnalyzer::HcalGeometryAnalyzer( const edm::ParameterSet& iConfig ) 
-    : m_label("_master")
-{
-    m_label = iConfig.getParameter<std::string>( "HCALGeometryLabel" );
+HcalGeometryAnalyzer::HcalGeometryAnalyzer( const edm::ParameterSet& iConfig ) : ps0(iConfig) {
+  useOld_ = iConfig.getParameter<bool>( "UseOldLoader" );
 }
 
-HcalGeometryAnalyzer::~HcalGeometryAnalyzer( void )
-{}
+HcalGeometryAnalyzer::~HcalGeometryAnalyzer( void ) {}
 
 void
-HcalGeometryAnalyzer::analyze( const edm::Event& /*iEvent*/, const edm::EventSetup& iSetup )
-{
-    edm::ESHandle<HcalTopology> topologyHandle;
-    iSetup.get<HcalRecNumberingRecord>().get( topologyHandle );
-    const HcalTopology* topology ( topologyHandle.product() ) ;
+HcalGeometryAnalyzer::analyze( const edm::Event& /*iEvent*/, const edm::EventSetup& iSetup ) {
 
-    edm::ESHandle<CaloSubdetectorGeometry> pG;
-    iSetup.get<HcalGeometryRecord>().get( HcalGeometry::producerTag() + m_label, pG );
-    
-    const CaloSubdetectorGeometry* caloGeom = pG.product();
-    const std::vector<DetId>& ids = caloGeom->getValidDetIds();
+  edm::ESHandle<HcalDDDRecConstants> hDRCons;
+  iSetup.get<HcalRecNumberingRecord>().get(hDRCons);
+  const HcalDDDRecConstants hcons = (*hDRCons);
+  edm::ESHandle<HcalTopology> topologyHandle;
+  iSetup.get<HcalRecNumberingRecord>().get( topologyHandle );
+  const HcalTopology topology = (*topologyHandle);
+
+  CaloSubdetectorGeometry* caloGeom(0);
+  if (useOld_) {
+    HcalHardcodeGeometryLoader m_loader(ps0);
+    caloGeom = m_loader.load(topology);
```

Comment:
@bsunanda - this a local variable. It's better to drop "m_" in the name. Please, see my previous comment.